docs/probes.md

```diff
@@ -26,105 +26,114 @@ Make sure to provide enough delay and failureThreshold for the VM and the agent
-        periodSeconds: 10
-        # Note that timeoutSeconds value does not have any impact before K8s v1.20.
-        timeoutSeconds: 5
       domain:
         cpu:
           cores: 1
           sockets: 1
           threads: 1
         devices:
           disks:
-            - disk:
+            - name: containerdisk
+              disk:
                 bus: virtio
-              name: cloudinitdisk
-            - bootOrder: 1
+            - name: cloudinitdisk
               disk:
                 bus: virtio
-              name: rootdisk
           interfaces:
-            - masquerade: {}
-              model: virtio
-              name: nic-0
-          networkInterfaceMultiqueue: true
+            - name: default
+              masquerade: {}
           rng: {}
         machine:
           type: pc-q35-rhel8.2.0
         resources:
           requests:
             memory: 1Gi
-      hostname: probe-test
       networks:
-        - name: nic-0
+        - name: default
           pod: {}
+      readinessProbe:
+        exec:
+          command: ["cat", "/tmp/healthy.txt"]
+        failureThreshold: 10
+        initialDelaySeconds: 120
+        periodSeconds: 10
+        # Note that timeoutSeconds value does not have any impact before K8s v1.20.
+        timeoutSeconds: 5
       terminationGracePeriodSeconds: 180
       volumes:
+        - containerDisk:
+            image: quay.io/containerdisks/fedora
+          name: containerdisk
         - cloudInitNoCloud:
-            userData: |
+            userData: |-
               #cloud-config
-              user: fedora
-              password: fedora
               chpasswd:
                 expire: false
+              password: password
+              user: fedora
               packages:
                 qemu-guest-agent
               runcmd:
-                - [ "systemctl", "enable", "--now", "qemu-guest-agent" ]
-                - [ "touch" "/tmp/ready.txt" ]
+                - ["touch", "/tmp/healthy.txt"]
+                - ["sudo", "chcon", "--type", "virt_qemu_ga_exec_t", "/tmp/healthy.txt"]
+                - ["sudo", "systemctl", "enable", "--now", "qemu-guest-agent"]
           name: cloudinitdisk
-        - containerDisk:
-            image: kubevirt/fedora-cloud-container-disk-demo
-          name: rootdisk
 ```
 2. Apply the VM manifest
 
 ```sh
-kubectl apply -f /tmp/probe-test.yaml
+kubectl apply --filename /tmp/probe-test.yaml
 ```
 
 3. Start the VM
 
 ```sh
-kubectl virt start probe-test
+virtctl start readiness-probe
 ```
 
 4. (optional) Watch the VM events in a separate shell
 
 ```sh
 # This will stream the events including any probe failures.
-# Observe the guest-agent becomming available here.
+# Observe the guest-agent becoming available here.
 kubectl get events --watch
 ```
 
+5. Check if the `.status.ready` field is `true`, it may take a bit
+
+```sh
+kubectl get virtualmachine readiness-probe --watch --output jsonpath='{.status.ready}{"\n"}'
```

Comment:
Could `kubectl wait vms/readiness-probe --for=condition=Ready` work here?

Reply:
Yes, it works. However, in my humble opinion, I think it's better for learners to spectate the .status.ready changes than just wait for it to meet the condition. What do you think?

Reply:
I understand what you're saying but that isn't common practice, using `wait` and `--for` is. I'll not block on this but I'd still rather use them over this command.

Reply:
I see. I think you are right, I've seen more often `--for`, so I will change it. Thanks!